Program.cs

```diff
@@ -1,6 +1,74 @@
 ﻿using System;
 using CodingCampusCSharpHomework;
 
+class Storage
+{
+    public bool LoadFromString(string descriptionString)
+    {
+        string[] fields = descriptionString.Split(';');
+        bool isParseSucceed = fields.Length == 4;
+
+        m_Name = fields[0];
+        m_Address = fields[1];
+        isParseSucceed = m_Coordinates.LoadFromString(fields[2], fields[3]);
+
+        if (!isParseSucceed)
+        {
+            Console.Error.WriteLine("Storage parse error.");
+        }
+
+        return isParseSucceed;
+    }
+
+    public float GetDistanceToPoint(Coordinates point)
+    {
+        return m_Coordinates.GetDistanceToPoint(point);
+    }
+
+    public string GetUserDescription()
+    {
+        return $"Name: {m_Name}; Address: {m_Address}";
+    }
+
+    private string m_Name;
+    private string m_Address;
+    private Coordinates m_Coordinates = new Coordinates();
+}
+
+class Coordinates
+{
+    public bool LoadFromString(string longitudeString, string latitudeString)
```

Comment:
Can be made static, then you won't have to create empty coordinates on line 35. If you want to avoid null checks, just make constructors for ```Coordinates``` and ```Storage``` private to force creation of those classes through the said static function. Kind of a matter of preference, I'm just not a fan of objects that don't hold any meaningful value and are just waiting to be initialized (with exceptions of course). So yeah, this all is optional.